xdsl/dialects/emitc.py

```diff
@@ -296,12 +305,100 @@ def is_supported_emitc_type(type_attr: Attribute) -> bool:
                 not isinstance(t, EmitC_ArrayType) and is_supported_emitc_type(t)
                 for t in type_attr.types
             )
-        case EmitC_PtrDiffT():
+        case EmitC_PtrDiffT() | EmitC_SignedSizeT() | EmitC_SizeT():
             return True
         case _:
             return False
 
 
+# Type constraint for EmitC operations
+class EmitCType(AttrConstraint):
+    """Constraint that verifies an attribute is a supported EmitC type."""
+
+    def verify(
+        self, attr: Attribute, constraint_context: ConstraintContext | None = None
+    ) -> None:
+        if not is_supported_emitc_type(attr):
+            raise VerifyException(f"Type {attr} is not a supported EmitC type")
+
+    def mapping_type_vars(
+        self, type_var_mapping: dict[TypeVar, AttrConstraint]
+    ) -> AttrConstraint:
+        # No type variables to map in this constraint
+        return self
+
+
+emitc_type = EmitCType()
+
+
+class EmitC_BinaryOp(IRDLOperation, abc.ABC):
```

Comment:
We use Operation as a convention for abstract superclasses
```suggestion
class EmitC_BinaryOperation(IRDLOperation, abc.ABC):
```